### DISCUSSION TITLE
Filter usability - better distinction between expandable groups and fields

### DISCUSSION THREAD

**Discussion by jdjoshuadavison** [directus/directus#19705] [Feature Requests]
### The overarching problem
Users find the search and filters difficult to use.
This is one of a number of requests to help solve this problem.

### Expandable groups and fields
Groups are expandable, but not selectable. This means you can click anywhere on the item, and it will expand. Some fields are also expandable, most notably relationship fields, but also any field with any special function (such as date fields), however these are often also selectable, which means if you click on the label, it selects it, and in order to expand this option you have to actually click the expand icon (arrow). It should be easy to tell the difference between something that is just expandable, and something that is both selectable and expandable. Each clickable element should be easily distinguishable.

### Motivation

Users find the filters difficult to use.

### Detailed Design

In this example, metadata is a group, so it is only expandable, not selectable, all the other expandable fields are also selectable:
![filter-ui-buttons](https://github.com/directus/directus/assets/34007858/234219f5-608c-4fae-af2b-de3f64194404)

https://github.com/directus/directus/assets/34007858/7bc4f2e3-5b68-44fc-86ea-294f77f8a52a

### Must Have:
- Each clickable element should be easily distinguishable. I.E. If the label and the expand arrows do different things, there should be clear distinction between them.

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

None

### Alternatives

This was just a quick mock up to illustrate one way that the different elements could be visually distinguished. Other visual designs would be possible.

### Adoption Strategy

Nothing to adopt - just enjoy a better UI!

**Comment by jdjoshuadavison** [comment]
Related #19706 
Related #19707

**Comment by rijkvanzanten** [comment]
@benhaynes Any design/ux feelings/concerns around adding an explicit visual signifier to the button if/when there's a difference between this "group expand only" versus "click parent or open nested"?

An alternative would be to make make the split even more obvious by having two "buttons" side by side in the same row for those like `[ User Created      ] [ > ]`

**Reply by benhaynes** [reply]
Good point! I think the proposed highlight above feels backward, as all of them can be expanded (the **right** arrow button), but only some can be selected (the **left** title area). I also think that there is a drawback here of visual clutter... which could be avoided by moving all of these areas to a hover on the row.

So I would propose what Rijk described above, having two distinct buttons: the left title and right arrow. But to only show these backgrounds on the hover of the row... and then highlight a hover state of the button when you're on one of the two. If it's a group that only has the right expand, then only that button would show.... but TBD on if that button is the whole width or just the arrow to be clear.

**Reply by jdjoshuadavison** [reply]
> I think the proposed highlight above feels backward, as all of them can be expanded (the right arrow button), but only some can be selected (the left title area).

Completely agree. I whipped this up very quickly at the end of the day on Friday, so it was really just to demonstrate some basic ideas, rather than an actual design suggestion.

> So I would propose what Rijk described above, having two distinct buttons: the left title and right arrow. But to only show these backgrounds on the hover of the row... and then highlight a hover state of the button when you're on one of the two. If it's a group that only has the right expand, then only that button would show.... but TBD on if that button is the whole width or just the arrow to be clear.

Yes, that all sounds very good!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
